### ISSUE TITLE
Tutorial Parsing Script fail; Is the program Hallucinating a code error?

### ISSUE THREAD

**Issue by Multi-A-Andi** [renpy/renpy#4859]
I just updated my RenPy Launcher to 8.1.1 because the Tutorial Suddenly broke on me, not able to start after I opened all scripts. (It worked perfectly normal the previous day)

This is the Traceback:
`File "game/tutorial_atl.rpy", line 483: Line with id tutorial_atl_da7a7759 appears twice. The other line is game/tutorial_atl.rpy:476`

I looked into the Code at the indicated Lines 483 and 476
 
```
476    e "Finally, an ATL block can be used as part of a show statement, instead of the at clause."
477
478
479    example:
480        show logo base:
481            yoffset 10
482
483    e "When ATL is used as part of a show statement, values of properties exist even when the transform is changed. So even though your click stopped the motion, the image remains in the same place." id tutorial_atl_da7a7759

```
There is no `id tutorial_atl_da7a7759` in line 476, as the traceback claims. What is happening?? I have no clue why the program is seemingly hallucinating code that doesn't exist?

**Comment by Gouvernathor** [comment]
Every line has an id, depending on a lot of factors when it isn't written explicitly. It's likely the natural id of the first line is the same as the manual id of the second. I'll look into it.

**Comment by Multi-A-Andi** [comment]
> Every line has an id, depending on a lot of factors when it isn't written explicitly. It's likely the natural id of the first line is the same as the manual id of the second. I'll look into it.

Thing is, this is all unedited, I downloaded the Launcher Update and immediately launched the Tutorial Program, without doing anything to the code. That's what is tripping me up.

**Comment by Gouvernathor** [comment]
I can't reproduce this, even after force recompiling the Tutorial, under the 8.1.1 Release.
What do you mean by "after I opened all scripts" exactly ?
Does the error pop just as you boot the tutorial, or rather when you reach the said lines ?

**Comment by Gouvernathor** [comment]
The code you copied is incorrect, the id clauses aren't that in the 8.1.1 version, the bug you're referring to has been fixed two years ago in 60c056c6aaf07566b944f252d708049febdb2440.
I imagine you may have had the file open during the update, or that something else corrupted the update process. In any case, close your text editors and update to 8.1.1 again.
You may do that either by updating to the nightly and coming back to the release, or by setting the `allow_repair_updates` variable to True in the launcher's console and updating to 8.1.1 in-place (if you don't know how to do that, do it the first way).

**Comment by Multi-A-Andi** [comment]
> I can't reproduce this, even after force recompiling the Tutorial, under the 8.1.1 Release. What do you mean by "after I opened all scripts" exactly ? Does the error pop just as you boot the tutorial, or rather when you reach the said lines ?

I used the "All scripts" button in the launcher for the script. that opened all of the tutorial scripts in my Editor because i wanted to look at something. I didn't touch anything and closed all of the scripts again then tried launching the Tutorial again but then it gave me an error. So i thought to update it, so that it would fix whatever happened. When I then Launched the Tutorial I got that exact error like i said in the initial issue report.

**Comment by Multi-A-Andi** [comment]
> The code you copied is incorrect, the id clauses aren't that in the 8.1.1 version, the bug you're referring to has been fixed two years ago in [60c056c](https://github.com/renpy/renpy/commit/60c056c6aaf07566b944f252d708049febdb2440). I imagine you may have had the file open during the update, or that something else corrupted the update process. In any case, close your text editors and update to 8.1.1 again. You may do that either by updating to the nightly and coming back to the release, or by setting the `allow_repair_updates` variable to True in the launcher's console and updating to 8.1.1 in-place (if you don't know how to do that, do it the first way).

Okay i updated again and it seems everything is fine. It may have been a hiccup with the update like you theorized.

**Comment by Org-gith** [comment]
Did you try to reinstall the sdk/make a clean install somewhere else, sometimes things can simply go wrong outside of Ren'py, AV could have blocked a file during the installation or some remains of a previous installation could conflict with the update if it was installed in the same folder. Or just Windows messing up something...
